transaction/core/src/input_rules.rs

```diff
@@ -53,6 +69,107 @@ impl InputRules {
                 return Err(InputRuleError::MissingRequiredOutput);
             }
         }
+
+        self.verify_partial_fill_rules(block_version, tx)?;
+
+        Ok(())
+    }
+
+    // Partial-Fill rules verification (MCIP #42)
+    fn verify_partial_fill_rules(
+        &self,
+        _block_version: BlockVersion,
+        tx: &Tx,
+    ) -> Result<(), InputRuleError> {
+        if let Some(fractional_change) = self.fractional_change.as_ref() {
+            // There is a fractional change output. Let's try to unblind its amount.
+            let fractional_change_amount = fractional_change.reveal_amount()?;
+            // Let's check if there is a corresponding real change output.
+            let real_change = tx
+                .prefix
+                .outputs
+                .iter()
+                .find(|x| fractional_change.tx_out.matches_ignoring_amount(x))
+                .ok_or(InputRuleError::MissingRealChangeOutput)?;
+            // Let's try to unblind its amount.
+            let real_change_amount =
+                try_reveal_amount(real_change, fractional_change.amount_shared_secret.as_ref())?;
+
+            // Check the bounds of the real change amount
+            if real_change_amount.token_id != fractional_change_amount.token_id {
+                return Err(InputRuleError::RealOutputTokenIdMismatch);
+            }
+            if real_change_amount.value > fractional_change_amount.value {
```

Comment:
It could happen I think if the `counterparty` used a bad `TransactionBuilder` instead of the real one? Nothing in the enclave verification code, except what we put in this function, prevents them from changing the value of that TxOut to anything they want when the make the corresponding "real" output